solidity/contracts/converter/types/liquidity-pool-v3/LiquidityPoolV3Converter.sol

```diff
@@ -0,0 +1,1356 @@
+      *
+      * @param _reserveId       reserve id
+      * @param _reserveBalance  reserve balance
+    */
+    function setReserveBalance(uint256 _reserveId, uint256 _reserveBalance) internal {
+        require(_reserveBalance <= MAX_UINT128, "ERR_RESERVE_BALANCE_OVERFLOW"); 
+        uint256 otherBalance = getReserveBalance(3 - _reserveId);
+        reserveBalances = (_reserveBalance << ((_reserveId - 1) * 128)) | (otherBalance << ((2 - _reserveId) * 128));
+    }
+
+    /**
+      * @dev syncs the stored reserve balance for a given reserve with the real reserve balance
+      *
+      * @param _reserveToken    address of the reserve token
+    */
+    function syncReserveBalance(IERC20Token _reserveToken) internal {
+        uint256 reserveId = reserveIds[_reserveToken];
+        uint256 balance = _reserveToken == ETH_RESERVE_ADDRESS ? address(this).balance : _reserveToken.balanceOf(address(this));
+        setReserveBalance(reserveId, balance);
+    }
+
+    /**
+      * @dev syncs all stored reserve balances
+    */
+    function syncReserveBalances() internal {
+        IERC20Token _reserveToken0 = reserveTokens[0];
+        IERC20Token _reserveToken1 = reserveTokens[1];
+        uint256 balance0 = _reserveToken0 == ETH_RESERVE_ADDRESS ? address(this).balance : _reserveToken0.balanceOf(address(this));
+        uint256 balance1 = _reserveToken1 == ETH_RESERVE_ADDRESS ? address(this).balance : _reserveToken1.balanceOf(address(this));
+        require(balance0 <= MAX_UINT128, "ERR_RESERVE_BALANCE_OVERFLOW"); 
+        require(balance1 <= MAX_UINT128, "ERR_RESERVE_BALANCE_OVERFLOW"); 
+        reserveBalances = balance0 | (balance1 << 128);
+    }
+
+    /**
+      * @dev helper, dispatches the Conversion event
+      *
+      * @param _sourceToken     source ERC20 token
+      * @param _targetToken     target ERC20 token
+      * @param _trader          address of the caller who executed the conversion
+      * @param _amount          amount purchased/sold (in the source token)
+      * @param _returnAmount    amount returned (in the target token)
+    */
+    function dispatchConversionEvent(
+        IERC20Token _sourceToken,
+        IERC20Token _targetToken,
+        address _trader,
+        uint256 _amount,
+        uint256 _returnAmount,
+        uint256 _feeAmount)
+        internal
+    {
+        // fee amount is converted to 255 bits -
+        // negative amount means the fee is taken from the source token, positive amount means its taken from the target token
+        // currently the fee is always taken from the target token
+        // since we convert it to a signed number, we first ensure that it's capped at 255 bits to prevent overflow
+        assert(_feeAmount < 2 ** 255);
+        emit Conversion(_sourceToken, _targetToken, _trader, _amount, _returnAmount, int256(_feeAmount));
+    }
+
+    /**
+      * @dev deprecated since version 28, backward compatibility - use only for earlier versions
```

Comment:
deprecated functions should move to the end of the contract

Reply:
Done